app/controllers/certificates_controller.rb

```diff
@@ -20,16 +20,15 @@ def show
 
   def new
     @certificate = Certificate.new(volunteer_id: @volunteer.id, user_id: current_user.id)
```

Comment:
you could probably use volunteer: @volunteer or directly build on the @volunteer association.